src/server/bnetserver/REST/LoginRESTService.cpp

```diff
@@ -288,11 +288,83 @@ int32 LoginRESTService::HandlePost(soap* soapClient)
 
         AddLoginTicket(loginResult.login_ticket(), std::move(accountInfo));
     }
+    else if (const uint32 maxWrongPassword = static_cast<uint32>(sConfigMgr->GetIntDefault("WrongPass.MaxCount", 0)))
+    {
+        PreparedStatement* stmt = LoginDatabase.GetPreparedStatement(LOGIN_SEL_BNET_ACCOUNT_ID_BY_EMAIL);
+        stmt->setString(0, login);
+
+        if (PreparedQueryResult result = LoginDatabase.Query(stmt))
+        {
+            const uint32 account_id = result->Fetch()[0].GetUInt32();
+
+            stmt = LoginDatabase.GetPreparedStatement(LOGIN_SEL_BNET_FAILED_LOGINS);
+            stmt->setUInt32(0, account_id);
+            const uint32 currentFailedLogins = LoginDatabase.Query(stmt)->Fetch()[0].GetUInt32();
+
+            if(sConfigMgr->GetIntDefault("WrongPass.Logging", 0))
+                TC_LOG_DEBUG("server.rest", "Account %s Id %u attempted to connect with wrong password", login.c_str(), account_id);
+
+            if (currentFailedLogins >= maxWrongPassword)
+            {
+                const enum BanMode banType = (enum BanMode)sConfigMgr->GetIntDefault("WrongPass.BanType", BAN_IP);
+                const int32 banTime = sConfigMgr->GetIntDefault("WrongPass.BanTime", 600);
+
+                if (banType == BAN_ACCOUNT)
+                    BanBnetAccount(account_id, banTime);
+                else
+                    BanIp(ip_address, banTime);
+
+                stmt = LoginDatabase.GetPreparedStatement(LOGIN_UPD_BNET_RESET_FAILED_LOGINS);
+                stmt->setUInt32(0, account_id);
+                LoginDatabase.AsyncQuery(stmt);
+            }
+            else
+            {
+                stmt = LoginDatabase.GetPreparedStatement(LOGIN_UPD_BNET_FAILED_LOGINS);
+                stmt->setUInt32(0, account_id);
+                LoginDatabase.Query(stmt);
+            }
+        }
```

Comment:
I could avoid the account_id query  in case of ip ban. And may be reset the failed logins when account or ip is unban.
 For the rest, I'm not sure.
I need the curent number of failed logins.
I need to check if there account or ip is already banned to avoid re-ban it.